app/client/pages/CreateListingPage.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Card from 'react-bootstrap/Card';
+import logo from '../../assets/images/logo/black_transparent.svg';
+import CreateListing from '../components/CreateListing.jsx';
+import Redirect from '../components/routing/Redirect.jsx';
+import Splash from '../components/layouts/Splash.jsx';
+import { withSession } from '../components/Remote.jsx';
+import selectors from './CreateListingPage.module.scss';
+
+/** Login */
+class CreateListingPage extends React.Component {
+  /** @param {Object} props */
+  constructor(props) {
+    super(props);
+    this.handleAddListing = this.handleAddListing.bind(this);
+  }
+
+  /** @param {Object} credentials */
+  handleAddListing(credentials) {
+    this.props.postRemote('/sessions', {
+      body: credentials,
+      success: (response) => this.props.updateSession(response.body || {})
+    });
+  }
+
+  /** @return {JSX.Element} */
+  render() {
+    return (
+      <>
+        {
+          (this.props.session?.user) ?
```

Comment:
I'm guessing this is copied from `Login.jsx` - you actually shouldn't need this guard here because this should be "private" route, whereas `Login.jsx` is "public" (see my suggestion in the `routes.mjs` file on how to specify that). Setting that flag will make it so that an unauthenticated user would be prompted with the login page when trying to visit this route.

However, this is actually convenient to have already with respect to the changes requested in the handler function - you repurpose this as

```suggestion
          (this.state.submit) ?
```